innovation_sweet_spots/utils/google_sheets.py

```diff
@@ -0,0 +1,83 @@
+import os.path
+import dotenv
+from innovation_sweet_spots import PROJECT_DIR, logging
+from pathlib import Path
+import oauth2client
+import google
+import pandas as pd
+from df2gspread import utils as d2g_utils
+from df2gspread import gspread2df as g2d
+from df2gspread import df2gspread as d2g
+import gspread
+
+
+def _is_valid_credentials(credentials):
+    """Function to monkey patch df2gspread.utils._is_valid_credentials."""
+    return isinstance(
+        credentials,
+        (oauth2client.client.OAuth2Credentials, google.oauth2.credentials.Credentials),
+    )
+
+
+def get_credentials_path() -> Path:
+    """Finds the path to the credentials file."""
+    if os.path.isfile(PROJECT_DIR / ".env"):
+        # Load the .env file
+        dotenv.load_dotenv(PROJECT_DIR / ".env")
+        try:
+            # Try fetching the path to credentials
+            return Path(os.environ["GOOGLE_SHEETS_CREDENTIALS"])
+        except:
+            pass
+    # If no .env file or if the key is not in the .env file
+    logging.warning("No credentials found!")
+    return None
+
+
+# Monkey patch function and constant
+d2g_utils._is_valid_credentials = _is_valid_credentials
+d2g_utils.CLIENT_SECRET_FILE = get_credentials_path()
```

Comment:
This updates `df2gspread.utils` so that:

- `_is_valid_credentials` accepts instances of both `oauth2client.client.OAuth2Credentials` and `google.oauth2.credentials.Credentials`.
- `CLIENT_SECRET_FILE` points to the client secrets file specified in `.env` rather than a fixed path in their utils.